packages/codegen-ui-react/lib/forms/react-form-renderer.ts

```diff
@@ -123,7 +124,14 @@ export abstract class ReactFormTemplateRenderer extends StudioTemplateRenderer<
 
     const wrappedFunction = this.renderFunctionWrapper(this.component.name, variableStatements, jsx, false);
 
-    const result = printer.printNode(EmitHint.Unspecified, wrappedFunction, file);
+    let result = printer.printNode(EmitHint.Unspecified, wrappedFunction, file);
+    if (this.componentMetadata.formMetadata?.onValidationFields) {
+      Object.entries(this.componentMetadata.formMetadata?.onValidationFields).forEach(
+        ([fieldName, validationRules]) => {
+          result = result.replace(`${fieldName}-validation-rules`, JSON.stringify(validationRules));
```

Comment:
Replaces distinct validation rules string with stringified validationRules array. I did this because validationRules array is already in the shape needed for the validation function